src/main/groovy/com/calgaryscientific/gradle/VeracodeWorkflowSandboxTask.groovy

```diff
@@ -0,0 +1,125 @@
+        build_version = project.findProperty("build_version")
+    }
+
+    VeracodeGetBuildInfoSandboxTask buildInfoSandboxTask = new VeracodeGetBuildInfoSandboxTask()
+    VeracodeCreateBuildSandboxTask createBuildSandboxTask = new VeracodeCreateBuildSandboxTask()
+    VeracodeUploadFileSandboxTask uploadFileSandboxTask = new VeracodeUploadFileSandboxTask()
+    VeracodeBeginPreScanSandboxTask beginPreScanSandboxTask = new VeracodeBeginPreScanSandboxTask()
+    VeracodeBeginScanSandboxTask beginScanSandboxTask = new VeracodeBeginScanSandboxTask()
+
+    void run() {
+        // Retrieve the properties in the run method to allow for testing
+        // TODO: Look for a better way
+        app_id = project.findProperty("app_id")
+        sandbox_id = project.findProperty("sandbox_id")
+        build_version = project.findProperty("build_version")
+        maxUploadAttempts = project.findProperty("maxUploadAttempts")
+        waitTimeBetweenAttempts = project.findProperty("waitTimeBetweenAttempts")
+        delete = project.findProperty("delete")
+        println "[debug] app_id: ${app_id}"
+        println "[debug] sandbox_id: ${sandbox_id}"
+        println "[debug] build_version: ${build_version}"
+
+        // Apply default properties to tasks
+        buildInfoSandboxTask.app_id = app_id
+        buildInfoSandboxTask.sandbox_id = sandbox_id
+        buildInfoSandboxTask.veracodeAPI = veracodeAPI
+
+        createBuildSandboxTask.app_id = app_id
+        createBuildSandboxTask.sandbox_id = sandbox_id
+        createBuildSandboxTask.build_version = build_version
+        createBuildSandboxTask.veracodeAPI = veracodeAPI
+
+        uploadFileSandboxTask.app_id = app_id
+        uploadFileSandboxTask.sandbox_id = sandbox_id
+        uploadFileSandboxTask.maxUploadAttempts = maxUploadAttempts
+        uploadFileSandboxTask.waitTimeBetweenAttempts = waitTimeBetweenAttempts
+        uploadFileSandboxTask.delete = delete
+        uploadFileSandboxTask.veracodeAPI = veracodeAPI
+
+        beginPreScanSandboxTask.app_id = app_id
+        beginPreScanSandboxTask.sandbox_id = sandbox_id
+        beginPreScanSandboxTask.veracodeAPI = veracodeAPI
+
+        beginScanSandboxTask.app_id = app_id
+        beginScanSandboxTask.sandbox_id = sandbox_id
+        beginScanSandboxTask.veracodeAPI = veracodeAPI
+
+        // Get build info
+        printf "[debug] getBuildInfo\n"
+        buildInfoSandboxTask.run()
+
+        // Save to variable to do the API call only once
+        String buildStatus = buildInfoSandboxTask.getBuildStatus()
+        printf "[debug] buildStatus: %s\n", buildStatus
+
+        // Done previous work
+        if (buildStatus == "Results Ready") {
+            printf "[debug] createBuild\n"
+            createBuildSandboxTask.run()
+            buildStatus = buildInfoSandboxTask.getBuildStatus()
+            printf "[debug] buildStatus: %s\n", buildStatus
+        }
+
+        // Clean build or with some uploaded files
```

Comment:
Again based on the comments, you are looking at extract the if block to something like
uploadFileAndBeginPreScanForCleanBuild?